### PR TITLE
Remove direct package references on System.Collections.Immutable and System.Reflection.Metadata

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Contributes to: https://github.com/dotnet/source-build/issues/4344

With https://github.com/dotnet/runtime/pull/100595, source-build got broken in release/8.0.1xx, due to package downgrade issue. Source-build consumes MSBuild 17.8.5, while runtime is on 17.8.3. Source-build will update MSBuild package references to live ones (17.8.5), but two runtime projects had direct package references, on S.C.I and S.R.M which caused package downgrade warning (prompted to error in source-build).

The fix is to remove these direct package references as they will be transitively resolved from MSBuild package reference. This will allow both source-build and repo build to get the correct sets of references.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Number.BigInteger.cs [hunk 1]
```diff
@@ -31,7 +31,9 @@ internal static partial class Number
             private const int MaxBits = BitsForLongestBinaryMantissa + BitsForLongestDigitSequence + BitsPerBlock;
 
             private const int BitsPerBlock = sizeof(int) * 8;
-            private const int MaxBlockCount = (MaxBits + (BitsPerBlock - 1)) / BitsPerBlock;
+
+            // We need one extra block to make our shift left algorithm significantly simpler
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Number.BigInteger.cs:35]: Unrelated changes?

Comment (NikolaMilosavljevic) [reply src/libraries/System.Private.CoreLib/src/System/Number.BigInteger.cs:35]: Yeah, closing this one - will open new PR - thanks!

**Comment by NikolaMilosavljevic** [comment]
Closing in favor of a new PR.
